mantidproject/mantid
# Add Python Profiling to Workbench

**Description of work.**

Adds the option to run workbench with python's `cProfile` module.

**To test:**
1. Run workbench from the command line with a `--profile path/to/output.prof` parameter with a path to where you would like the output of the profiling to be placed. 
2. Install the `snakeviz` module from pip and run it with the output file, by doing something like ` python3 -m snakeviz ~/path/to/output/file.prof `
3. `snakeviz` should use the profile and open a browser window that allows the profile to be analysed in a graphical format. 
3. Check the state of the profile is reasonable.

Refs #28499 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
